### DISCUSSION TITLE
Add conditional post-upgrade helm hook to run initcontainer with airflow variables import

### DISCUSSION THREAD

**Discussion by y0zg** [apache/airflow#30393] [Ideas] [answered]
Hello, 

I want to pre-install some variables (which are available in UI) and I know this isn't something can be done with helm chart. Do you mind if I add such functionality using helm hook? Or there is already some thing similar?

Thanks, 
OK

**Comment by potiuk** [comment, marked as answer]
As I explained in slack already, Helm is not a good place to update database variables (not in general case. As explained - replacing the variables (that could have been modified manually) in the helm chart every time you install a new version of helm might override those variables, so doing it automatically at helm install is not a good idea (unless you find a good solution for that). This has been discussed already https://github.com/apache/airflow/discussions/16387 so you might read it before you go any further.

Also using hooks alredy causes prpblems in platforms like ArgoCD, Flux or Terraform that do not support hooks, so implementing a general solution there (on top of the above) using hooks is likely not a good idea: https://airflow.apache.org/docs/helm-chart/stable/index.html#installing-the-chart-with-argo-cd-flux-or-terraform

But - if after reading all that you are able to provide a solution that addresses all those problem already discussed - surely. Open PR and we might discuss it when it's there (but all those problems and questions will be raised an your PR should address all of them before merged).

Also note that in case you REALLY want to use hooks for YOUR case, you can always create your own simple chart and make airflow a subchart of your chart https://helm.sh/docs/chart_template_guide/subcharts_and_globals/ - then in your own "parent" chart you can add whatever you want - without making it "generic" and solving all the problems we are anticipating - our helm chart needs to be generic-enough, while your chart might address only your case. This is standard way on how you can add functionality to exisitng charts that will address your specific needs.

**Comment by y0zg** [comment]
Thanks @potiuk for your support! I solved my query with usage of `extraInitContainers` and scheduler args `args: ["bash", "-c", "airflow variables import /opt/airflow/variables.json && exec airflow scheduler"]`. I know not ideal but at this moment it suits my needs
